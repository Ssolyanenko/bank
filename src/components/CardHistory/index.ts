export * from './History';
export * from './Pagination';
export * from './CardHistory';
