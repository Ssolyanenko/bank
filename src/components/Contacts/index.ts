export * from './Contacts';
export * from './ContactsFooter';
export * from './ContactsInformation';
