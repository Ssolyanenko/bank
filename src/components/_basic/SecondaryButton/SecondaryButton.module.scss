@import 'src/styles/variables.module.scss';

.secondaryBtn {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  width: 100%;
  min-height: 56px;
  color: $grayDark_6;
  background: $brand_5;
  border-radius: 2px;
  font-style: normal;
  font-weight: $bold;
  font-size: 16px;
  line-height: 16px;
  border: 1px solid $brand_5;
  cursor: pointer;
  box-sizing: border-box;
  text-align: center;
  transition: 0.5s;
  margin-top: 32px;

  &:hover {
    border-color: $brand_4;
    background-color: $brand_4;
  }

  &:disabled {
    cursor: not-allowed;
    background-color: $grayLight_1;
    border-color: $grayLight_1;
    color: $grayDark_7;
  }

  &.small {
    max-width: 143px;
  }

  &.medium {
    max-width: 171px;
  }

  &.large {
    max-width: 343px;
  }
}
