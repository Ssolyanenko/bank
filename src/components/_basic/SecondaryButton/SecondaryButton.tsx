import { FC } from 'react';

import { ButtonType, Size } from 'interfaces/common/componentsSettings';
import classes from './SecondaryButton.module.scss';

interface Props {
  children: React.ReactNode;
  name?: string;
  onClick?(): void;
  className?: string;
  type?: ButtonType;
  isDisabled?: boolean;
  size?: Size;
}

export const SecondaryButton: FC<Props> = ({
  children,
  onClick,
  name = 'secondaryButton',
  className = '',
  isDisabled = false,
  type = ButtonType.BUTTON,
  size = Size.MEDIUM,
}) => (
  <button
    className={`${classes.secondaryBtn} ${className} ${classes[size]}`}
    type={type}
    onClick={onClick}
    name={name}
    disabled={isDisabled}
  >
    {children}
  </button>
);
