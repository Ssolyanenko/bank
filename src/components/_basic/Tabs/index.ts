export * from './Tabs';
export * from './TabTitle';
