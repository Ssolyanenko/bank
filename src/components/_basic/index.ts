export * from './PrimaryButton';
export * from './Button';
export * from './Input';
export * from './InputPassword';
export * from './ModalWrapper';
export * from './ModalConfirmation';
export * from './PhoneCodeTimer';
export * from './ModalBox';
export * from './CloseButton';
export * from './BackButton';
export * from './StepperForm';
export * from './StepperLabel';
export * from './ChatButton';
export * from './BanTimer';
export * from './CardTemplate';
export * from './SecondaryButton';
export * from './generic';
export * from './SelectStyled';
export * from './TextFieldStyled';
export * from './InputStyled';
export * from './CardTemplate';
export * from './Modal';
export * from './SecureWrapper';
export * from './InputRange';
export * from './DropDown';
export * from './RadioButtonsGroup';
export * from './FormError';
export * from './Checkbox';
export * from './Switcher';
export * from './EditButton';
export * from './CardBlock';
export * from './Tabs';
export * from './ModalSuccess';
export * from './DropDownWithCheckbox';
export * from './SelectFormikTemplate';
export * from './Loader';
export * from './DatePicker';
export * from './CustomRoutes';
