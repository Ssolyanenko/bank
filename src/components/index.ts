export * from './LandingPageComponents';
export * from './Forms';
export * from './ArrowsBlock';
export * from './RootModal';
export * from './Slider';
export * from './Map';
export * from './Contacts';
export * from './MainPageComponents';
export * from './ExchangeRates';
export * from './CardProducts';
export * from './CardHistory';
export * from './SetPinSuccessModal';
export * from './MyCardsApplications';
export * from './MyLoans';
export * from './MyCardsApplications';
export * from './InformationBlock';
export * from './StatusTag';
export * from './DropDownWithSearchBranches';
export * from './FormInputBox';
export * from './MyCards';
export * from './ActivateBlockCardModal';
export * from './UserAccount';
export * from './SetPinCodeFormModal';
export * from './CardDetails';
export * from './CardStatement';
export * from './CardStatementButtons';
export * from './Accordion';
export * from './ReceiptButtons';
export * from './TemporaryComponent/TemporaryComponent';
export * from './TransferModal';
export * from './ModalChoosePaymentCard';
export * from './CardInfo';
export * from './ShareReceiptModal/ShareReceiptModal';
export * from './SocialMedia';
export * from './ResetFormsConfirmModal';
export * from './TransactionLimit';
export * from './ChangeTransactionLimit';
export * from './ActivateBlockCard';
export * from './CardManage';
export * from './ChangePINCode';
export * from './ResponseModal';
export * from './TransactionLimit';
