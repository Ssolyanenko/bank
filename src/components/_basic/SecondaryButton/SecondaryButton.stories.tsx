import { ComponentMeta, ComponentStory } from '@storybook/react';

import { SecondaryButton } from './SecondaryButton';

export default {
  title: 'Basic/Buttons/SecondaryButton',
  component: SecondaryButton,
  parameters: {
    docs: {
      description: {
        component: 'Button to confirm',
      },
    },
  },
} as ComponentMeta<typeof SecondaryButton>;

const Template: ComponentStory<typeof SecondaryButton> = (props) => <SecondaryButton {...props} />;

export const Default = Template.bind({});

Default.args = {
  onClick(): void {
    alert('click');
  },
  children: 'Secondary Button',
};
