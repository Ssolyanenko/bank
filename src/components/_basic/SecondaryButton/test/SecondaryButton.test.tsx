import React from 'react';
import { shallow } from 'enzyme';

import { SecondaryButton } from 'components/_basic';
import { Default } from 'components/_basic/SecondaryButton/SecondaryButton.stories';

describe('<SecondaryButton />', (): void => {
  test('should make a snapshot', (): void => {
    const props = {
      children: 'test',
      onClick: jest.fn(),
      className: 'button',
    };
    const wrapper = shallow(<SecondaryButton {...props} />);
    const component = wrapper.getElement();

    expect(component).toMatchSnapshot();
  });

  test('should render SecondaryButton from storybook', (): void => {
    const props = {
      children: 'test',
      onClick: jest.fn(),
      className: 'button',
    };
    const wrapper = shallow(<Default {...props} />);

    expect(wrapper.find(SecondaryButton).length).toBe(1);
  });
});
