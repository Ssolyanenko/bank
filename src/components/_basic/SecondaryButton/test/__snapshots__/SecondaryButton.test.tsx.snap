// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecondaryButton /> should make a snapshot 1`] = `
<button
  className="secondaryBtn button medium"
  disabled={false}
  name="secondaryButton"
  onClick={[MockFunction]}
  type="button"
>
  test
</button>
`;
